Home_Work_27/HW27.cpp

```diff
@@ -0,0 +1,574 @@
+#include <iostream>
+#include <vector>
+#include <thread>
+#include <chrono>
+#include <mutex>
+#include <algorithm>
+#include <string>
+
+std::mutex threadMutex;
+
+std::vector<int> generateVector(int size);
+void sumElements(std::vector<int> inVector, int startIndex, int endIndex, int& sumResult);
+
+int main()
+{
+	std::vector<int> vector10000 = generateVector(10000);
+	std::vector<int> vector100000 = generateVector(100000);
+	std::vector<int> vector1000000 = generateVector(1000000);
+	std::vector<int> vector10000000 = generateVector(10000000);
+	std::vector<int> vector100000000 = generateVector(100000000);
+
+	int resultOfSum =0;
+
+	std::cout << std::endl;
+	std::cout << "PC Threads: " << std::thread::hardware_concurrency();
+	std::cout << std::endl;
+	std::cout << "PC Threads 2X: " << std::thread::hardware_concurrency()*2;
+	std::cout << std::endl;
+
+	std::cout << std::endl;
+	std::cout << "Single Thread";
+	std::cout << std::endl;
+
+	auto begin = std::chrono::high_resolution_clock::now();
+
+	std::thread tread1(sumElements, vector10000, 0, 9999, std::ref(resultOfSum));
+	tread1.join();
+
+	auto end = std::chrono::high_resolution_clock::now();
```

Comment:
the case with no additional threads should be measured without the time overload of creating a thread. That is without new threads but directly in main thread.